### DISCUSSION TITLE
Support multisite with subdirectories in `navigate` function

### DISCUSSION THREAD

**Discussion by SantosGuillamot** [WordPress/gutenberg#53104] [Interactivity API]
As reported [here](https://github.com/WordPress/wp-movies-demo/issues/46), some sites use a subdirectory, so using a relative path in the `navigate` function won't work as expected. Imagine I have a multisite in `mysite.com/en/`. If I call `navigate("/")`, it will navigate to `mysite.com` instead of `mysite.com/en/`.

Not sure what is the best way to handle this, but I assume the `navigate` function should be aware of this somehow. Should we expose the site URL in the store as discussed [here](https://github.com/WordPress/block-interactivity-experiments/issues/160)? Should we populate the site URL in the SSR using the [`get_site_url()` function](https://developer.wordpress.org/reference/functions/get_site_url/)?

**Comment by luisherranz** [comment]
Can we rely on the [`<base>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base) element? Is it always present in WordPress? If not, maybe we should populate `state.router.base`?

**Comment by SantosGuillamot** [comment]
> Can we rely on the [<base>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base) element? Is it always present in WordPress?

I think we can't rely on the `<base>` tag. I've checked a bunch of WordPress sites, and none of them had it.

> If not, maybe we should populate `state.router.base`?

That was my initial idea. I guess it will be helpful in more use cases.

What I am not sure about is if this should be populated from the server using the [`get_site_url()` function](https://developer.wordpress.org/reference/functions/get_site_url/) or somehow in the client. Or maybe both? If `state.router.base` doesn't exist, populate it with `window.location.origin` or something similar. I'm thinking in edge cases where this is used outside of WordPress, and that function is unavailable.

**Comment by luisherranz** [comment]
I think you can't get it reliably from `window.location`, can you? I.e., imagine that you land at `https://site.com/a/b/c`. There's no way to know if WordPress is installed in `/`, `/a` or `/a/b`.

**Comment by SantosGuillamot** [comment]
> I think you can't get it reliably from window.location, can you? I.e., imagine that you land at https://site.com/a/b/c. There's no way to know if WordPress is installed in /, /a or /a/b.

Yes, I think you can't. That's why I suggested using it as a fallback in case it is not defined from the server.

**Comment by westonruter** [comment]
I don't think `get_site_url()` is quite right since it is where WordPress is installed, not what the actual site address is. The site address is the _home_ URL returned by `get_home_url()`. I suppose outputting that in a custom `meta` tag is what is needed. Alternatively, a [rel-home microformat](https://microformats.org/wiki/rel-home) could be used (which doesn't seem fully standardized, but it does have modest usage).

```php
add_action( 'wp_head', function () {
   printf( '<link rel="home" href="%s">', esc_url( get_home_url() ) ); 
} );
```
